### ISSUE TITLE
Document translations process

### ISSUE THREAD

**Issue by ghostwords** [EFForg/privacybadger#1280]
To make it as easy as possible for everyone to participate in translating Privacy Badger. Start by documenting existing process (which involves mucking about with JSON files). Should strongly consider utilizing a service (we should already have a Transifex account) to ease participation.

Following up on https://github.com/EFForg/privacybadger/pull/1272#issuecomment-289485777.

**Comment by dvd2000** [comment]
@ghostwords Privacy Badger already has a Transifex account [here](https://www.transifex.com/eff/privacy-badger/). Some people (including me) have asked to join as translators but only the maintainers can accept us. One of them is @cooperq, I've just sent a mail to him. If you want I can try to move the translation files to Transifex but I need to be added as a Project Maintainer in the Transifex.

**Comment by ghostwords** [comment]
Cool, I'd like to give Transifex another try, seems like it will simplify the process for translators. Could you CC me on the e-mail? I'm alexei at eff dot org.

**Comment by dvd2000** [comment]
@ghostwords I've sent you the email

**Comment by jony0008** [comment]
I have finished the Swedish translation on Transifex, I hope you can merge it if it has not been done yet.

**Comment by ghostwords** [comment]
Have you seen our existing [Swedish translations](https://github.com/EFForg/privacybadger/blob/ec90ae2504e52e448b10fb85f4f8bd0d8b1f679e/src/_locales/sv/messages.json)?

**Comment by jony0008** [comment]
Yes, it is not complete

**Comment by ghostwords** [comment]
@jony0008 Please see updates to Swedish translations in #1417.

**Comment by cowlicks** [comment]
related: https://github.com/EFForg/privacybadger/pull/1193#issuecomment-282830814

**Comment by ghostwords** [comment]
Current process is something like:

When I learn somehow there are new translations on Transifex:
1. Fetch the updates using [transifex-client](https://docs.transifex.com/client/installing-the-client), by running `tx pull -l sv` (for example, "sv" is language code for Swedish).
2. Verify the changes as much as I can with automated translation tools.
3. Commit, push, and open a PR on GitHub.

When there are new translations on GitHub:
1. Verify the changes as much as I can with automated translation tools.
2. Merge the PR.
3. Fetch the latest for that language from Transifex and confirm these are the only changes, to avoid overwriting work on Transifex.
4. Sync the changes with Transifex by running `tx push -t -l sv` (for example).

If we haven't ever synced GitHub and Transifex before for a language:
1. Verify that the only changes between GitHub and Transifex are key order and additional blank "description" properties on the Transifex side.
2. Commit and push the Transifex version to GitHub as the definitive version to ease the syncing process going forward for that language.

**Comment by dvd2000** [comment]
@ghostwords What should be the preferred translation method: GitHub or Transifex?
At this moment the CONTRIBUTING page says that translations are handled via GitHub. Should we change this and use Transifex to translate Privacy Badger?

**Comment by ghostwords** [comment]
Yeah, this issue is about expanding on the `Translations are currently done on GitHub. Here is a great example of a translation pull request. See the translations label for related issues and more example pull requests.` bit in the CONTRIBUTING doc.

I am not sure ... Transifex might be easier for translators to work with, but requires signing up for Transifex. Some existing translations contributors seem fine using GitHub directly. I think we should recommend whatever is easier for translators, but accept updates via either channel.

**Comment by dvd2000** [comment]
Ok, I will try to write something in the CONTRIBUTING doc to ease the translation process for new contributors

**Comment by dvd2000** [comment]
@ghostwords  I have some questions to ask you to get a better understanding of the translation process so that I can write better instructions about it.

Do you receive some kind of notification when new translations are added on Transifex or we should find a way for the translator to notify you (or someone else)  when he adds or updates a translation?
Are you the only person who can pull Transifex translations or it can be done by anyone? 

Maybe we could let the translators use Transifex to handle the translation (if they are already used to the platform) and then instruct them on how to download their work and open a pull request here on GitHub. In this way you won't need to keep track of the work on Transifex and you should only update it when you see a PR on GitHub.

At this moment you are the person working on the GitHub/Transifex syncing process, can you continue to do it by yourself or do you need someone else to help you with this work? Should translators contact you if they need some help or maybe just tag you in translations PR's?

The main problem we have in using the GitHub+Transifex approach is to try to avoid double work (two translators working on the same translation) so we should keep the translations on the two platforms updated and avoid having to wait too long for a PR to be merged or for the Transifex translation to be updated. Should the translators ask before starting to work on a translation or should they check by themselves if some work is currently being done on their language?

**Comment by ghostwords** [comment]
So far I've learned of new translations on Transfex because I've been personally notified by the translator. Looking at Transifex settings, I see that I should get notified when "a file has been 100% translated in one of its target languages", which is too high of a bar, I think. I don't see any option that will notify me of changes soon after they happen, regardless of completion percentage. Users can message me on Transifex and I'll get notified that way though.

**Comment by ghostwords** [comment]
Transifex translations can be pulled by anyone with a Transifex account (I think), but pushed only by collaborators or maybe only by coordinators. I haven't tested this yet, just some guesses.

**Comment by ghostwords** [comment]
>Maybe we could let the translators use Transifex to handle the translation (if they are already used to the platform) and then instruct them on how to download their work and open a pull request here on GitHub. In this way you won't need to keep track of the work on Transifex and you should only update it when you see a PR on GitHub.

That would work, and make things a little easier for Privacy Badger developers, but I'm OK making things easier for translators too. As in, no GitHub account, no problem.

**Comment by ghostwords** [comment]
I'm the person who does the syncing for now. It doesn't take much time to do it, not a big deal. Also, I monitor GitHub activity regardless, no need to tag me to get my attention.

**Comment by ghostwords** [comment]
Good point about avoiding double work. If somebody works on a language on Transifex and doesn't notify us, and then somebody else works on that language on GitHub, we'll have duplication of effort. It would help if Transifex could automatically notify me of work being done on a language.

**Comment by dvd2000** [comment]
Ok, so translators working on Transifex should notify you of their work while there is no need of notification from translators on GitHub?

Also, don't we need some kind of revision process to assure a high quality of the translations? Maybe just having at least second translator proofreading and correcting errors or waiting some fixed minimum time before updating the translation on GitHub to allow some reviews?

**Comment by ghostwords** [comment]
Correct, if you work on Transifex, send us a message on Transifex when you start working and when your changes are ready to be pulled in. GitHub, no need, we get notified automatically.

I would say having a review process is a nice-to-have, a refinement we can add as we go along. We don't really have enough translators involved to swing that at this point, I think.

**Comment by ghostwords** [comment]
We do basic sanity checks using things like automated translation tools, of course.

**Comment by dvd2000** [comment]
Ok, I will open a pull request to write some stuff to help translators.
We have two options:
1) Write everything in the already existing CONTRIBUTING.md file (but maybe it would be a bit too much and I think it's better to keep this file short)
2) Write a dedicated file with instructions for translators (probably in the doc folder) and link to it from the CONTRIBUTING file.
I think we should prefer the 2nd option to keep the CONTRIBUTING file simple and short.

**Comment by ghostwords** [comment]
Yep, agreed, the second option is better.

**Comment by ghostwords** [comment]
#1626 was actually merged. Thanks, @dvd2000! Sorry, not sure what I did wrong to break GitHub's merge detection.
